lib/spack/spack/test/config.py

```diff
@@ -1157,6 +1158,19 @@ def test_bad_path_double_override(config):
             pass
 
 
+def test_license_dir_config(mutable_config, mock_packages):
+    """Ensure license directory is customizable"""
+    assert spack.config.get("config:license_dir") == spack.paths.default_license_dir
+    assert spack.package.Package.global_license_dir == spack.paths.default_license_dir
+    assert spack.repo.get("a").global_license_dir == spack.paths.default_license_dir
+
+    rel_path = os.path.join(os.path.sep, "foo", "bar", "baz")
```

Comment:
This doesn't create a relative path... I'm not sure that matters, but the naming is a little odd.